src/libraries/Common/tests/System/Security/Cryptography/AlgorithmImplementations/SlhDsa/SlhDsaTests.cs

```diff
@@ -0,0 +1,83 @@
+            Assert.Throws<ArgumentNullException>(() => SlhDsa.ImportSlhDsaPublicKey(null, ReadOnlySpan<byte>.Empty));
+            Assert.Throws<ArgumentNullException>(() => SlhDsa.ImportSlhDsaSecretKey(null, ReadOnlySpan<byte>.Empty));
+            Assert.Throws<ArgumentNullException>(() => SlhDsa.ImportSlhDsaPrivateSeed(null, ReadOnlySpan<byte>.Empty));
+        }
+
+        [Theory]
+        [MemberData(nameof(AlgorithmsData))]
+        public static void ArgumentValidation(SlhDsaAlgorithm algorithm)
+        {
+            using SlhDsa slhDsa = SlhDsa.GenerateKey(algorithm);
+
+            Assert.Throws<ArgumentNullException>(() => slhDsa.ExportEncryptedPkcs8PrivateKey(ReadOnlySpan<byte>.Empty, null));
+            Assert.Throws<ArgumentNullException>(() => slhDsa.ExportEncryptedPkcs8PrivateKeyPem(ReadOnlySpan<byte>.Empty, null));
+            Assert.Throws<ArgumentNullException>(() => slhDsa.ExportEncryptedPkcs8PrivateKeyPem(ReadOnlySpan<char>.Empty, null));
+            Assert.Throws<ArgumentNullException>(() => slhDsa.TryExportEncryptedPkcs8PrivateKey(ReadOnlySpan<byte>.Empty, null, Span<byte>.Empty, out _));
+            Assert.Throws<ArgumentNullException>(() => slhDsa.TryExportEncryptedPkcs8PrivateKey(ReadOnlySpan<char>.Empty, null, Span<byte>.Empty, out _));
+
+            int publicKeySize = algorithm.PublicKeySizeInBytes;
+            int secretKeySize = algorithm.SecretKeySizeInBytes;
+            int privateSeedSize = algorithm.SecretKeySizeInBytes / 4 * 3;
+
+            Assert.Throws<CryptographicException>(() => SlhDsa.ImportSlhDsaPublicKey(algorithm, new byte[publicKeySize - 1]));
+            Assert.Throws<CryptographicException>(() => SlhDsa.ImportSlhDsaPublicKey(algorithm, new byte[publicKeySize + 1]));
+            Assert.Throws<CryptographicException>(() => SlhDsa.ImportSlhDsaSecretKey(algorithm, new byte[secretKeySize - 1]));
+            Assert.Throws<CryptographicException>(() => SlhDsa.ImportSlhDsaSecretKey(algorithm, new byte[secretKeySize + 1]));
+            Assert.Throws<CryptographicException>(() => SlhDsa.ImportSlhDsaPrivateSeed(algorithm, new byte[privateSeedSize - 1]));
+            Assert.Throws<CryptographicException>(() => SlhDsa.ImportSlhDsaPrivateSeed(algorithm, new byte[privateSeedSize + 1]));
+
+            Assert.Throws<ArgumentException>(() => slhDsa.ExportSlhDsaPublicKey(new byte[publicKeySize - 1]));
+            Assert.Throws<ArgumentException>(() => slhDsa.ExportSlhDsaSecretKey(new byte[secretKeySize - 1]));
+            Assert.Throws<ArgumentException>(() => slhDsa.ExportSlhDsaPrivateSeed(new byte[privateSeedSize - 1]));
+
+            // Context length must be less than 256
+            Assert.Throws<ArgumentOutOfRangeException>(() => slhDsa.SignData(ReadOnlySpan<byte>.Empty, Span<byte>.Empty, new byte[256]));
+            Assert.Throws<ArgumentOutOfRangeException>(() => slhDsa.VerifyData(ReadOnlySpan<byte>.Empty, Span<byte>.Empty, new byte[256]));
```

Comment:
Like with the above (and I won't mention it again after this), since these require an instance they're either a) TPM boundary tests (exempt from the IsSupported condition), or b) instance tests (locally exempt from the condition but realistically require it); but not the same as "argument validation on the static methods" tests

Reply:
Separated out into:
- TPM boudary tests (SlhDsaBaseClassTests)
- Static method tests (SlhDsaConstructionTests) - some of these also don't need the platform support check so I just moved the XUnit conditional attribute down to the relevant methods.
- TODO: Instance tests (once OpenSsl/Windows is implemented)